pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/v2/builder/MultipleTreesBuilder.java

```diff
@@ -81,9 +84,10 @@ public MultipleTreesBuilder(List<StarTreeV2BuilderConfig> builderConfigs, File i
     _builderConfigs = builderConfigs;
     _buildMode = buildMode;
     _segmentDirectory = SegmentDirectoryPaths.findSegmentDirectory(indexDir);
+    _indexDirectory = indexDir;
     _metadataProperties =
         CommonsConfigurationUtils.fromFile(new File(_segmentDirectory, V1Constants.MetadataKeys.METADATA_FILE_NAME));
-    Preconditions.checkState(!_metadataProperties.containsKey(MetadataKey.STAR_TREE_COUNT), "Star-tree already exists");
+    _existingStarTrees = getExistingStarTrees();
```

Comment:
We might not want to always separate all the trees. Ideally the separator can track all the star-tree metadata (merge `SeparatedStarTreesMetadata` into the separator), and only extract the tree that can be reused into the target folder. Currently we always copy all the trees into separate files, even though they might not be reusable. This way we don't need the temp folder as well.

Reply:
Once all the trees are extracted I am deleting the existing star-tree index files so that new index can be built. Even if we just extract the required tree on demand instead of separating all trees at the start, we need to copy the existing index files to a temp folder for the separator to operate. So we can't avoid the temp folder, but I can surely update the logic to only separate the required tree on demand if current logic feels like an overhead.

Reply:
Oh I see the point. Basically we need to first move the existing files into a temporary location, then delete it when the new trees are created